### ISSUE TITLE
Possible installation issue with websockets, proxy, tomcat in vagrant box

### ISSUE THREAD

**Issue by chezhouse** [freeciv/freeciv-web#321]
I've installed freeciv-web through the vagrantfile method, with one exception.
In the vagrantfile, instead of using 80 and 443 for the ports in the windows section, I used 8080 and 8443, akin to the linux/mac setup.
When directing a chrome/firefox browser to localhost:8080, the normal staging screens appear as expected.
However, when attempting to get the client server communication hooked up, I encounter an abnormal socket closure - 1006.
This occurs right after the send within check_websocket_ready().
The network log displays the 7000 message with the response status 101 for the switch protocol upgrade that is resultant from the creation of the websocket.

when checking the status (sudo /vagrant/scripts/status-freeciv-web.sh), I noticed that
pubstatus, mailstatus, proxy direct, proxy nginx, websocket direct, websocket nginx, and freeciv-web were all down.

when attempting to start it up again (sudo /vagrant/scripts/start-freeciv-web.sh),

```
$ sudo /vagrant/scripts/start-freeciv-web.sh
Starting up Freeciv-web: nginx, tomcat, publite2, freeciv-proxy.
Starting nginx first.
nginx already running!
Starting up Tomcat
sudo: /var/lib/tomcat9/bin/catalina.sh: command not found
Starting publite2
Will sleep for 8 seconds, then do a status test...
starting meta-stats.py
Starting Freeciv-Earth-mapgen.
Publite2 started
Starting Freeciv-PBEM
unable to start meta-stats
unable to start freeciv-earth-mapgen
unable to start pbem
  File "publite2.py", line 29, in <module>
    from pubstatus import *
  File "/vagrant/publite2/pubstatus.py", line 2, in <module>
    from tornado import web, ioloop, httpserver
ModuleNotFoundError: No module named 'tornado'
Checking that Freeciv-web is running correctly... (-v for verbose)

Checking nginx                                     OK!
Checking Tomcat                                    OK!
Checking freeciv-web on Tomcat                     OK!
Checking Tomcat DB connection                      OK!
Checking Pubstatus                                 Down
Checking Mailstatus                                Down
Checking freeciv-proxy directly                    Down
Checking freeciv-proxy through nginx               Down
Checking WebSocket directly (freeciv-proxy)        Down
Checking WebSocket through nginx                   Down
Checking freeciv-web (spawned by publite2)         Down
Checking webclient.min.js generation               OK!
Checking tileset generation                        OK!

--------------------------------
Check of Freeciv-web is done!
```

This is from the vagrant box install

**Comment by andreasrosdalw** [comment]
One problem here is that Tornado isn't installed correctly. This can be seen in your logs here:

File "publite2.py", line 29, in <module>
    from pubstatus import *
  File "/vagrant/publite2/pubstatus.py", line 2, in <module>
    from tornado import web, ioloop, httpserver
ModuleNotFoundError: No module named 'tornado'

So Tornado isn't installed. See https://www.tornadoweb.org/en/stable/

https://github.com/freeciv/freeciv-web/blob/d307cca05e2caac43632b31ba6d4970b92cee703/scripts/install/install.sh#L224
https://github.com/freeciv/freeciv-web/blob/d307cca05e2caac43632b31ba6d4970b92cee703/freeciv-proxy/requirements.txt



Why did you change the port? This will likely also cause problems.

In your installation directory, there will be a log file called install.log. Can you please attach this file here?

**Comment by chezhouse** [comment]
Thanks for you response.   I see that Tornado wasn't installed correctly, too. 
I am using the _vagrantfile_ (and corresponding box), so there wasn't a clear indication that I needed to do that separately.

On the port front, the reason I"m using 8080 and 8443 is because of a forwarding port collision.  Windows 10 has a Windows Update agent that grabs info on 80/443, so I wanted to reroute.  

I believe there is no current problem in hindsight.
After I cleared out all of the .vagrant and .virtualbox entries in my home directory, the repository built fine through the standard process.  I did have to change the ports to 8080 and 8443 and increased the boot timeout `config.vm.boot_timeout`.  In addition, I performed
`vagrant plugin install vagrant-vbguest` prior to the `vagrant up` command.
After that, the build process seemed to work fine and I was able to launch the site, create a new game with websockets working as expected.

Thanks for you attention and help.
